### ISSUE TITLE
Projector <=> deriveContextObjectType

### ISSUE THREAD

**Issue by tekacs** [sangria-graphql/sangria#176]
I've tried taking a quick look about to see how possible this is, but I was wondering - is it possible to refactor `Projector` in such a way that it can be passed to methods generated by  `deriveContextObjectType` in the same way [that Context is](https://github.com/sangria-graphql/sangria/blob/513021f0bc56696f0aac9b616947baea30aba6e2/src/main/scala/sangria/macros/derive/DeriveObjectTypeMacro.scala#L150)?

As it stands, where Projector overrides the 'resolve' function, I can't immediately see a way to make this work short of refactoring its mechanism.

**Comment by OlegIlyenko** [comment]
Yeah, unfortunately there is no out-of-the-box support. One thing that I can recommend you to check is a `ReplaceField` macro setting. Here is an example:

```scala
val tpe = deriveObjectType[Ctx, TestSubject](
  ReplaceField("foo", Field("foo", ListType(StringType), resolve = Projector((ctx, proj) ⇒ {
    // resolve logic goes here
  }))))
```

**Comment by tekacs** [comment]
Thanks for this! Would you be opposed to a pull request refactoring the way Projector` works, if it were possible to make both the current API and this approach work?

**Comment by OlegIlyenko** [comment]
It would be cool if you could check it out! At the moment I'm not quite sure how to signal that particular field needs a projection (it's probably only relevant for methods, since the projection will be passed as an argument). Also projection should accept an argument (the depth). 

If you have ideas, I would be happy to discuss them. Especially how the end-user API would look like.
